hack/pipeline/main.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+func getSemaphoreDirs(dir string) ([]string, error) {
+	var dirs []string
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() && filepath.Base(path) == ".semaphore" {
+			dirs = append(dirs, path)
+		}
+		return nil
+	})
+	return dirs, err
+}
+
+func getYAMLFiles(dir string) ([]string, error) {
+	var files []string
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		// Skip the YAML .semaphore/semaphore.yml.d directory
```

Comment:
We want code comments to be providing additional context - it's obvious from the code that this line skips the semaphore.yml.d directory. The important thing the comment should explain is _why_ the code skips that directory.